!article/article.md

```diff
@@ -0,0 +1,170 @@
+
+This makes sure the repository is either using the built-in security mechanism in nuget.org that cryptographically validates the dependencies you’re using (when using CPM), or that a lock file is used to do the same during build time.
+
+### Examples:
+
+The following CLI commands will be marked as a pinned dependency:
+
+```
+nuget restore -LockedMode
+dotnet.exe restore --locked-mode
+msbuild.exe /t:restore /p:RestoreLockedMode=true
+```
+
+The following commands will be marked as unpinned dependencies:
+
+```
+nuget restore
+dotnet.exe restore
+msbuild.exe /t:restore
+```
+
+Unless you either:
+- set `ManagePackageVersionsCentrally` to true in the `directory.*.props` file AND add a specific version to all included packages
+- set the` RestorePackagesWithLockFile` attribute to true in all .csproj files in the repository.
+
+In case your project receives a low score on the pinned dependencies check, you can try the following remediation steps:
+
+- If your project is producing an application, declare all your dependencies with specific versions in your package format file (e.g. packages.config for NuGet).
+- If the package manager supports lock files (e.g. packages.lock.json  for NuGet), check these in the source code. The files maintain signatures for the entire dependency tree and prevent future exploitation in case the package is compromised.
+- For Dockerfiles, pin dependencies by hash as in this example. If using a manifest list, pin to the manifest list hash instead.
+- In GitHub workflows used in building and releasing your project, pin dependencies by hash. See main.yaml for example. To determine the permissions needed for your workflows and pin actions to commit SHAs, you may use StepSecurity's online tool.
+- To update dependencies after pinning, use tools such as Dependabot and Renovate bot.
+
+# An analysis of Scorecard checks for .NET open source repositories on GitHub
+
+On GitHub, there are over 1,000 C# and F# repositories with more than 1,500 stars. Our team analyzed the Scorecard scores for these repositories using the public Scorecard API to get an ecosystem-wide view of the state of .NET open-source projects. 
+
+The diagram below displays the average scores for all Scorecard checks for those repositories. Please note that scores below zero are not relevant and may indicate an error during the Scorecard check execution.
+
+![average](./average-scores.png)
+
+It’s especially noteworthy to look at the most critical security check, Dangerous Workflows. In this case, the check either receives a perfect score of 10 or identifies an error, which occurs in over 250 repositories. This reveals existing security gaps and highlights opportunities for improvement to strengthen overall security.
```

Comment:
is this a security gap or a support gap that highlights an opportunity to improve by contributing?

Reply:
Support gap, I reworded it.